src/components/input/input.scss

```diff
@@ -125,20 +152,14 @@ md-input-container {
     overflow: hidden;
     text-overflow: ellipsis;
     white-space: nowrap;
-    width: 100%;
-    order: 1;
     pointer-events: none;
     -webkit-font-smoothing: antialiased;
-    @include rtl(padding-left, $input-container-padding + 1px, 0);
-    @include rtl(padding-right, 0, $input-container-padding + 1px);
     z-index: 1;
-    transform: translate3d(0, $input-label-default-offset + 4, 0) scale($input-label-default-scale);
-    transition: transform $swift-ease-out-duration $swift-ease-out-timing-function;
-
-    // The max-width is necessary, because in some browsers, using this together with
-    // a calc might cause it to overflow the parent. See #7403
-    max-width: 100%;
-
+    font-size: inherit;
+    will-change: transform, font-size; // prep animation for smoothness
+    transform: translate3d(0, $input-label-default-offset, 0);
+    transition: transform $swift-ease-out-duration $swift-ease-out-timing-function,
+                font-size $swift-ease-out-duration $swift-ease-out-timing-function;
```

Comment:
I originally set it to `position: relative`, but I ran into the issue of it modifying messing around with width of the component. I couldn't figure out a way to get label to only affect layouts vertically and not horizontally.

`label` has `position: absolute` so it won't affect layout and I just set a really tall margin (now padding in my code) on the input container.

I was more worried about if browsers supported animation of `font-size`. I haven't tested this rebase, but the original one worked fined in Safari, IE10-Edge, Firefox, and Chrome. Color me surprised about that IE 10 compatibility.